jetty-core/jetty-server/src/main/java/org/eclipse/jetty/server/internal/HttpChannelState.java

```diff
@@ -353,23 +353,54 @@ public Runnable onIdleTimeout(TimeoutException t)
         {
             if (LOG.isDebugEnabled())
                 LOG.debug("onIdleTimeout {}", this, t);
+
+            // if we are currently demanding, take the onContentAvailable runnable to invoke below.
+            Runnable invokeOnContentAvailable = _onContentAvailable;
+            _onContentAvailable = null;
+
+            // If a write call is in progress, take the writeCallback to fail below
+            Runnable invokeWriteFailure = _response.lockedFailWrite(t);
+
+            // If an IO operation was in progress
+            if (invokeOnContentAvailable != null || invokeWriteFailure != null)
+            {
+                // Then we will just fail the current operation and all future IO operation.
+                // No listeners are called at all, as the application is already notified
+                // TODO should we make this read failure transient, so it can be ignored by the application?
+                if (_failure == null)
+                {
+                    _failure = Content.Chunk.from(t);
+                }
+                else if (ExceptionUtil.areNotAssociated(_failure.getFailure(), t) && _failure.getFailure().getClass() != t.getClass())
+                {
+                    _failure.getFailure().addSuppressed(t);
+                }
+
+                return _serializedInvoker.offer(invokeOnContentAvailable, invokeWriteFailure);
```

Comment:
Much of this was copied over from onFailure, so if there is an IO operation in progress, we just fail them when an idle timeout occurs.

I think this needs careful review